clang-tools-extra/clang-tidy/cppcoreguidelines/ProTypeConstCastCheck.cpp

```diff
@@ -14,13 +14,42 @@ using namespace clang::ast_matchers;
 
 namespace clang::tidy::cppcoreguidelines {
 
+static bool hasConstQualifier(QualType Type) {
+  const QualType PtrType = Type->getPointeeType();
+  if (!PtrType.isNull())
+    return hasConstQualifier(PtrType);
+
+  return Type.isConstQualified();
```

Comment:
In cppreference, `const_cast` means converting between types with different cv-qualification.
So `isConstQualified` is not enough, `isVolatileQualified` is also needed.